src/contracts/atlas/AtlasVerification.sol

```diff
@@ -548,27 +549,21 @@ contract AtlasVerification is EIP712, DAppIntegration, AtlasConstants {
     function _verifyUser(
         DAppConfig memory dConfig,
         UserOperation calldata userOp,
+        bytes32 userOpHash,
         address msgSender,
         bool isSimulation
     )
         internal
         returns (ValidCallsResult)
     {
+        if (userOp.from == address(this) || userOp.from == ATLAS || userOp.from == userOp.control) {
+            return ValidCallsResult.UserFromInvalid;
+        }
+
         // Verify the signature before storing any data to avoid
         // spoof transactions clogging up dapp userNonces
 
-        bool isFromContract = userOp.from.code.length > 0;
-
-        bool signatureValid = false;
-        if (isFromContract) {
-            if (userOp.from == address(this) || userOp.from == ATLAS || userOp.from == userOp.control) {
-                return ValidCallsResult.UserFromInvalid;
-            }
-            signatureValid = IAccount(userOp.from).validateUserOp(userOp, _getUserOpHash(userOp), 0) == 0;
-        } else {
-            // user is an EOA
-            signatureValid = _verifyUserSignature(userOp);
-        }
+        bool signatureValid = SignatureChecker.isValidSignatureNow(userOp.from, _hashTypedDataV4(_getUserOpHash(userOp)), userOp.signature);
```

Comment:
```suggestion
        bool signatureValid = SignatureChecker.isValidSignatureNow(userOp.from, _hashTypedDataV4(userOpHash), userOp.signature);
```
Related to @jj1980a's comment above, you probably intended to use the new `userOpHash` param here. That would save gas on needing to re-hash the userOp, and also already have checked if `getUserOperationHash()` or `getAltOperationHash()` are needed.

Reply:
On that note, shouldn't `getUserOperationPayload()` also take a `bool useAltOperationHash` param to allow it to return the hash a user would sign when `allowsTrustedOpHash()` is true?